### ISSUE TITLE
Device monitor should use gudev instead of plain udev

### ISSUE THREAD

**Issue by phako** [Jeija/gtkterm#22]
With GUDev you get main loop integration and GLib signals, there is no need for a separate thread.

**Comment by wvdakker** [comment]
Thanks, this is on my todo list.

**Comment by kev1pic** [comment]
Hi,

I would like to make some tests using gudev as suggested by phako. I'm new to meson build system and I'm facing the following issue.

I installed libgudev-1.0 and added dependcy in meson.build:
`gudev_deps = dependency('gudev-1.0', version: '>= 230')`

Dependency is correctly found:
`Run-time dependency gudev-1.0 found: YES 230`

But it seems that include and lib are not added:
`../src/gtkterm.c:37:25: fatal error: gudev/gudev.h: No such file or directory
 #include <gudev/gudev.h>` 

From build.ninja file:
`
build src/25a6634@@gtkterm@exe/gtkterm.c.o: c_COMPILER ../src/gtkterm.c || data/resource.h
DEPFILE = src/25a6634@@gtkterm@exe/gtkterm.c.o.d
ARGS = -Isrc/25a6634@@gtkterm@exe -Isrc -I../src -I. -I../ -Idata -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/u    sr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/incl    ude/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetyp    e2 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I    /usr/include/vte-2.91 -I/usr/include/p11-kit-1 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -g -p    thread
`

Any ideas what I could have missed ?

**Comment by Jeija** [comment]
You probably just forgot to add `gudev_deps` in the `dependencies` list in `gtkterm/src/meson.build`, it should look like this:
```
executable(
	'gtkterm', sources,
	export_dynamic : true,
	dependencies : [
		gtk_deps,
		vte_deps,
		udev_deps,
		gudev_deps,
		config
	],
	install : true
)
```

**Comment by Jeija** [comment]
Fixed by #28
